operator/main.go

```diff
@@ -159,43 +160,44 @@ func runOperator(cmd *cobra.Command) {
 	}
 
 	var (
-		nodeManager *ipam.NodeManager
+		nodeManager *allocator.NodeEventHandler
 		err         error
 	)
-
-	switch option.Config.IPAM {
-	case option.IPAMENI:
-		ipamAllocatorAWS, providerBuiltin := allocatorProviders["aws"]
+	if ipamMode := option.Config.IPAM; ipamMode != "" {
+		alloc, providerBuiltin := allocatorProviders[ipamMode]
```

Comment:
Previous version looked in `allocatorProviders` on presumption that it's either `option.IPAMENI` or `option.IPAMAzure`. It's possible for `option.Config.IPAM` to have other values, so the new version will fail as it will try to look for those values in the map, or have you removed all deprecated values as part of this PR?